src/go/rdctl/pkg/shutdown/shutdown.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shutdown
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"time"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/factoryreset"
+	"github.com/sirupsen/logrus"
+)
+
+func FinishShutdown(verbose bool) error {
+	switch runtime.GOOS {
+	case "darwin":
+		doCheckWithTimeout(checkProcessQemu, pkillQemu, 15, 2, "qemu", verbose)
+		doCheckWithTimeout(checkProcessDarwin, pkillDarwin, 5, 1, "the app", verbose)
+	case "linux":
+		doCheckWithTimeout(checkProcessQemu, pkillQemu, 15, 2, "qemu", verbose)
+		doCheckWithTimeout(checkProcessLinux, pkillLinux, 5, 1, "the app", verbose)
+	case "windows":
+		doCheckWithTimeout(checkProcessWindows, factoryreset.PowershellKillWindows, 15, 2, "the app", verbose)
+	default:
+		return fmt.Errorf("unhandled runtime: %s", runtime.GOOS)
+	}
+	return nil
+}
+
+func doCheckWithTimeout(checkFunc func(bool) bool, killFunc func(), retryCount int, retryWait int, operation string, verbose bool) {
+	for iter := 0; iter < retryCount; iter++ {
+		if iter > 0 {
+			if verbose {
+				fmt.Fprintf(os.Stderr, "checking %s showed it's still running; sleeping %d seconds\n", operation, retryWait)
```

Comment:
Any particular reason we're not just using `logrus` here (and elsewhere)?

Reply:
No, and using it cut down on propagation of the `verbose` flag.